content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm

```diff
@@ -20,24 +20,74 @@ includeLatestJQuery('sakai_filepicker_attach.vm');
 
 <script type="text/javascript">
 	<!--
+
+	const ONE_KB = 1024;
+	const maxFileSize = $!maxUploadFileSize * ONE_KB  * ONE_KB;
+
 	function openCopyrightWindow(theURL,winName,winSettings) 
 	{ 
 		window.open(theURL,winName,winSettings);
 		return false;
 	}
-	function submitform(id)
+	    
+	function formatFileSize(bytes) {
+		if(bytes == 0) return '0 Bytes';
+		var k = 1000,
+			sizes = ['Bytes', 'KB', 'MB', 'GB'],
+			i = Math.floor(Math.log(bytes) / Math.log(k));
+		return parseFloat((bytes / Math.pow(k, i)).toFixed(1)) + ' ' + sizes[i];
+	}
+
+
+	function checkUploadFileSize() {
+		const uploadFileSize = $('#upload')[0].files[0].size;
+		// return true if upload size is within the limit else returns false
+		return (uploadFileSize <= maxFileSize)
+	}
+
+	function submitform(id,needToCheckFileSize = false)  // default to not require file size check
 	{
 		var theForm = document.getElementById(id);
-		if(theForm && theForm.onsubmit)
-		{
-			theForm.onsubmit();
+		if (!needToCheckFileSize) { 
+			if(theForm && theForm.onsubmit)
+			{
+				theForm.onsubmit();
+			}
+			if(theForm && theForm.submit)
+			{
+				theForm.submit();
+			}
 		}
-		if(theForm && theForm.submit)
-		{
-			theForm.submit();
+		else {
+			if (checkUploadFileSize()) {
+				showNotif('submitnotifxxx','attachButton','attachForm');
```

Comment:
We should use something more meaningful here; `submitNotification` instead of `submitnotixxx`.

Reply:
suggestion completed